.gitlab-ci.yml

```diff
@@ -5,34 +5,44 @@ stages:
   - build
   - test
 
-# disabled because cannot find -luv -lcap when building
+# disabled because gcc9 can't build net_if.hpp
 #build:linux:
 #  image: alpine:latest
 #  tags:
 #    - linux
 #  stage: build
 #  before_script:
-#    - apk add --update g++ make cmake linux-headers libcap-dev libuv-dev
+#    - apk add --update g++ make cmake linux-headers libcap-dev libuv-dev curl-dev git
 #  script:
-#    - make STATIC_LINK=ON
+#    - make STATIC_LINK=ON DOWNLOAD_SODIUM=ON
 #  artifacts:
 #    paths:
 #      - "lokinet"
 
 # we'll just try our travis set up for now
-#build:linux_release:
-#  image: ubuntu:xenial
-#  tags:
-#    - linux
-#  stage: build
-#  before_script:
-#    - apt-get update && apt-get install -y binutils-gold build-essential bsdmainutils ca-certificates cmake curl git libcap-dev libcurl4-openssl-dev libsodium-dev libuv1-dev ninja-build
-#  script:
-#    - DOWNLOAD_SODIUM=ON STATIC_LINK=OFF make release
-#  artifacts:
-#    paths:
-#      - "build/daemon/lokinet"
-#      - "build/daemon/lokinetctl"
+build:linux_release:
+  image: ubuntu:xenial
+  tags:
+    - linux
+  stage: build
+  before_script:
+    - apt-get update && apt-get install -y binutils-gold build-essential bsdmainutils ca-certificates cmake curl git libcap-dev pkg-config wget xz-utils libuv1-dev libsodium-dev libcurl4-openssl-dev
+  script:
+    - make static DOWNLOAD_SODIUM=ON
+    # alternatively: git describe --tag
```

Comment:
I don't think that would work - most of the time won't be building a tag.

Reply:
It was just an option @despair86 recommended. It seemed to grab a valid tag set from my experimentation used with it. Returns `v0.6.2-38-gc6fa014a` for my branch currently (which isn't tagged). Found a webpage that describes it as 
> `git describe` command finds the most recent tag that is reachable from a commit

and 

> --tags
> Instead of using only the annotated tags, use any tag found in refs/tags namespace. This option enables matching a lightweight (non-annotated) tag.

from https://git-scm.com/docs/git-describe#Documentation/git-describe.txt---tags

I'd suggest to move the version number into a text file that the build system can inject like the motto. This would simplify version management for build pipelines. @despair86  said that would be fine with his Windows requirements too. But needs further discussion/buy-in with the rest of the team.

Reply:
That's rather interesting, I had no idea that command existed.